components/sst_importer/src/config.rs

```diff
@@ -47,12 +52,48 @@ impl Config {
             self.stream_channel_window = default_cfg.stream_channel_window;
         }
         if self.memory_use_ratio > 0.5 || self.memory_use_ratio < 0.0 {
+            return Err("import.mem_ratio should belong to [0.0, 0.5].".into());
+        }
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct ConfigManager(pub Arc<RwLock<Config>>);
+
+impl ConfigManager {
+    pub fn new(cfg: Config) -> Self {
+        ConfigManager(Arc::new(RwLock::new(cfg)))
+    }
+}
+
+impl online_config::ConfigManager for ConfigManager {
+    fn dispatch(&mut self, change: online_config::ConfigChange) -> online_config::Result<()> {
+        info!(
+            "import config changed";
+            "change" => ?change,
+        );
+
+        let mut cfg = self.rl().clone();
+        cfg.update(change)?;
+
+        if let Err(e) = cfg.validate() {
             warn!(
-                "import.mem_ratio should belong to [0.0, 0.5], change it to {}",
-                default_cfg.memory_use_ratio,
+                "import config changed";
+                "change" => ?cfg,
             );
-            self.memory_use_ratio = default_cfg.memory_use_ratio;
+            return Err(e);
         }
+
+        *self.wl() = cfg;
         Ok(())
     }
 }
+
+impl std::ops::Deref for ConfigManager {
```

Comment:
It's generally not recommended to use Deref as it's not easy to trace the code, per my understanding. 
I think in this case just use self.0.as_ref() directly seems more straightforward.